File: app/core/boards.py
"""Syllabus board constants.

Boards are tracked as plain strings on `School.board` and `Subject.board`
rather than a Board model — small enough that an enum / table would be
over-engineering. The list lives here so signup validation, subject
creation endpoints, and the frontend dropdown all share one source.

Adding a new board (e.g. when CISCE materials get loaded) is two-step:
  1. Append to `VALID_BOARDS` here, push backend.
  2. Frontend rebuild picks it up via `GET /me` or a future
     `GET /curriculum/boards` endpoint.
"""

# Order matters for UI presentation — most-common-first so the dropdown
# default behaviour ("just press enter on the highlighted option") gives
# the right answer for most signups.
VALID_BOARDS: tuple[str, ...] = (
    "CBSE",
    "NIOS",
    "ICSE",
    "State Board",
    "Other",
)


def is_valid_board(value: str) -> bool:
    """Case-sensitive membership check. The frontend dropdown uses these
    exact strings, so any inbound payload that doesn't match one of them
    is a client bug we should reject loudly."""
    return value in VALID_BOARDS

File: frontend/src/lib/boards.ts
/**
 * Syllabus boards supported by the platform. Mirrors
 * `app/core/boards.py` on the backend — adding a new board is a paired
 * change in both files. Order is the dropdown's display order so most
 * common (CBSE) sits at the top.
 */
export const VALID_BOARDS = [
  "CBSE",
  "NIOS",
  "ICSE",
  "State Board",
  "Other",
] as const;

export type Board = (typeof VALID_BOARDS)[number];
